File: Problemset/0477-Total Hamming Distance/Solution.rb
# @param {Integer[]} nums
# @return {Integer}
def total_hamming_distance(nums)
    ret = 0

    for i in 0...30
        zeros, ones = 0, 0
        for num in nums
            if (1 << i) & num == 0
                zeros += 1
            else
                ones += 1
            end
        end
        ret += zeros * ones
    end

    return ret
end

File: Problemset/0477-Total Hamming Distance/Solution.rs
impl Solution {
    pub fn total_hamming_distance(nums: Vec<i32>) -> i32 {
        let mut ret = 0;

        for i in 0..30 {
            let mut zeros = 0;
            let mut ones = 0;
            for num in &nums {
                match (1 << i) & num {
                    0 => zeros += 1,
                    _ => ones += 1,
                }
            }
            ret += zeros * ones;
        }

        ret
    }
}
